### PR TITLE
Wait until WFR goes Ready for Progress & Delete operations. Override …

### PR DESCRIPTION
**Pull request by NateThornton**
Wait until WFR goes Ready for Progress & Delete operations. 
Override with `--nowait`
Custom timeout in seconds with `-t` or `--timeout`

Signed-off-by: Nate Roiger <nate.roiger@hpe.com>

### CODE REVIEW

File: pkg/DWSUtility.py [hunk 1]
```diff
@@ -368,14 +368,14 @@ def do_resource_list(self):
 
             try:
                 crd_obj = self.dws.get_custom_resource_definition(crd)
-            except:
+            except Exception:
```
Comment (bdevcich) [inline pkg/DWSUtility.py:371]: Should use `DWSError` here. And we don't care if it fails?

**Review by roehrich-hpe — APPROVED** [review]

File: pkg/DWSUtility.py [hunk 2]
```diff
@@ -368,14 +368,14 @@ def do_resource_list(self):
...
                 continue
             printer_cols = self.dws.get_crd_printer_columns(crd_obj)
```
Comment (bdevcich-hpe) [inline pkg/DWSUtility.py:373]: Shouldn't this be above in the try/except block since `crd_ojb` could be none here?

File: pkg/DWSUtility.py [hunk 3]
```diff
@@ -368,14 +368,14 @@ def do_resource_list(self):
...
 
             plural, _, group = crd.partition(".")
             try:
                 resources = self.dws.list_cluster_custom_object(plural, group, apiver)
```
Comment (bdevcich) [inline pkg/DWSUtility.py:377]: Same here?

File: pkg/DWSUtility.py [hunk 4]
```diff
@@ -368,14 +368,14 @@ def do_resource_list(self):
...
-            except:
+            except Exception:
```
Comment (bdevcich) [inline pkg/DWSUtility.py:378]: Same as above.

File: pkg/DWSUtility.py [hunk 5]
```diff
@@ -684,6 +684,11 @@ def do_delete_wfr(self, name):
             wfr_list.append(name)
 
         for wfr_name in wfr_list:
+            if self.config.wait:
+                wfr = self.dws.wfr_get(wfr_name)
```
Comment (bdevcich) [inline pkg/DWSUtility.py:688]: Seems slim, but is there any chance the get returns an exception here?

File: pkg/Dws.py [hunk 6]
```diff
@@ -307,16 +308,16 @@ def remove_custom_resource_finalizers(self, crd_name):
...
         if isinstance(crd_name, list):
             crd_name = crd_name[0]
-        
+
         plural, _, group = crd_name.partition(".")
 
         try:
             crd_obj = self.get_custom_resource_definition(crd_name)
-        except:
+        except Exception:
```
Comment (bdevcich) [inline pkg/Dws.py:316]: Use `DWSError` here too.

**Review by matthew-richerson — APPROVED** [review]

File: pkg/DWSUtility.py (hunk 1 above)
Comment (NateThornton) [reply pkg/DWSUtility.py:371]: This and others are correcting a Flake8 error (linter). DWSError is a type of Exception

File: pkg/DWSUtility.py (hunk 2 above)
Comment (bdevcich) [inline pkg/DWSUtility.py:373]: Should this be in the try? It seems like `crd_obj` could be `None` without an exception thrown: https://github.com/kubernetes-client/python/blob/e3c0042e0fa69d72d5e4a9dfd481a0f896a67b66/kubernetes/client/api_client.py#L194.

File: pkg/DWSUtility.py (hunk 5 above)
Comment (NateThornton) [reply pkg/DWSUtility.py:688]: the get will throw if not found 
https://github.com/NearNodeFlash/dwsutil/blob/d04f654e1d94e593fe503ee23db3e1faf9ad996b/pkg/Dws.py#L469

**Review by ajfloeder — APPROVED** [review]

File: pkg/DWSUtility.py (hunk 2 above)
Comment (NateThornton) [reply pkg/DWSUtility.py:373]: `get_custom_resource_definition` will throw if `crd_name` doesn't exist.

File: pkg/DWSUtility.py (hunk 1 above)
Comment (bdevcich) [reply pkg/DWSUtility.py:371]: Doesn't flake8 complain that this is generic Exception handling? That's why I pointed this out. https://www.flake8rules.com/rules/E722.html

`except:` and `except Exception:` are equivalent in my mind, but maybe flake8 is ok with this.

File: pkg/DWSUtility.py (hunk 5 above)
Comment (bdevcich) [reply pkg/DWSUtility.py:688]: Ah, I see that `run()` has this function wrapped in try/except.

**Review by bdevcich — APPROVED** [review]

File: pkg/DWSUtility.py (hunk 1 above)
Comment (NateThornton) [reply pkg/DWSUtility.py:371]: Well the warning went away - so I went with it.

**Pull request merged**